src/ReservoirPriceOracle.sol

```diff
@@ -194,6 +201,8 @@ contract ReservoirPriceOracle is IPriceOracle, IReservoirPriceOracle, Owned(msg.
         rResults = new uint256[](aQueries.length);
 
         OracleAverageQuery memory lQuery;
+        // TODO: We create the `aQueries` array just to iterate through it. Can
+        // we not just do each call inline without allocating an array?
```

Comment:
okay not sure if I'm understanding correctly, but are you suggesting that we change the function signature to be of a single struct (not an array)

If that's the case, then during an `updatePrice` with several intermediate prices, `getTimeWeightedAverage` would have to be called multiple times, which increases the gas cost since it's a public function